### DISCUSSION TITLE
Hugo pipelines for scss processing

### DISCUSSION THREAD

**Discussion by lingej** [Lednerb/bilberry-hugo-theme#339] [Q&A]
In previous versions, Hugo pipelines were used to create the css files. Now this is done via npm.

If possible, I would like to do without the use of npm. Why are the Hugo pipelines no longer used?

Greetings Jörg

**Comment by igor-baiborodine** [comment]
@Lednerb, I think you have better intrinsic knowledge to answer this question.

**Reply by Lednerb** [reply]
Hey @lingej 

you're right, in previous versions hugo pipelines were used but with very limited functionality because all of the needed files were saved in the repository by simply doing a manual copy&paste job.

Therefore the dependencies were not managed through a package manager like `npm`, `gulp` or `webpack` etc.

Also the hugo pipelines had the dependency to the `extended` version of hugo that caused some confusion while using the theme with the normal hugo version.

Therefore I've decided to switch to `npm` and `webpack` some years(?) ago, so dependency updates were possible via `npm update`.
___

As the time has changed many things, I would try to stick with the hugo pipelines if I would decide today, because of all the potential problems `npm` comes along with, such as the huge mass of vulnerable, outdated or manipulated dependencies.

Nowadays hugo also comes with the `resources.GetRemote` functionality that can simplify a lot of things.

___

If someone submits a nice PR that also allows the customization features that are implemented today it would be possible to switch back to Hugo pipelines.

**Comment by Lednerb** [comment]
Hey @lingej 

you're right, in previous versions hugo pipelines were used but with very limited functionality because all of the needed files were saved in the repository by simply doing a manual copy&paste job.

Therefore the dependencies were not managed through a package manager like `npm`, `gulp` or `webpack` etc.

Also the hugo pipelines had the dependency to the `extended` version of hugo that caused some confusion while using the theme with the normal hugo version.

Therefore I've decided to switch to `npm` and `webpack` some years(?) ago, so dependency updates were possible via `npm update`.
___

As the time has changed many things, I would try to stick with the hugo pipelines if I would decide today, because of all the potential problems `npm` comes along with, such as the huge mass of vulnerable, outdated or manipulated dependencies.

Nowadays hugo also comes with the `resources.GetRemote` functionality that can simplify a lot of things.

___

If someone submits a nice PR that also allows the customization features that are implemented today it would be possible to switch back to Hugo pipelines.
